OwnYourData/ClinicalTrials/ClinicalTrialsView.swift

```diff
@@ -6,47 +6,96 @@
 // SPDX-License-Identifier: MIT
 //
 
+import CoreLocation
 import OpenAPIClient
 import SpeziViews
 import SwiftUI
 
-
 struct ClinicalTrialsView: View {
     @State private var viewState: ViewState = .idle
     @State private var trials: [TrialDetail] = []
+    @State private var collapseStates: [Bool] = [] // Track collapsed state for each trial
+    @State private var isLoading: Bool = false // Track whether trials are being actively loaded
```

Comment:
We could use the `viewState` property to track if the view is loading or not. Alternatively we can remove that property and only tack it using the boolean state. Advantage of using view state would be the automatic error injection if needed to display an alert using `viewStateAlert` modifier.